.github/workflows/deploy.yml

```diff
@@ -11,21 +11,28 @@ jobs:
     steps:
       - name: Checkout code
         uses: actions/checkout@v2
-      
-      # Use GitHub Actions' cache to shorten build times and decrease load on servers
-      - uses: actions/cache@v2
+
+      - uses: actions/setup-python@v2
         with:
-          path: vendor/bundle
-          key: ${{ runner.os }}-gems-${{ hashFiles('**/Gemfile') }}
-          restore-keys: |
-            ${{ runner.os }}-gems-
+          python-version: '3.x' # Version range or exact version of a Python version to use, using SemVer's version range syntax
+          architecture: 'x64'
 
-      - name: Build
-        uses: helaili/jekyll-action@v2
+      - uses: ruby/setup-ruby@v1
         with:
-          token: ${{ secrets.GITHUB_TOKEN }}
-          build_only: true
-          jekyll_build_options: -d out
+          ruby-version: 2.6 # Not needed with a .ruby-version file
+          bundler-cache: true # runs 'bundle install' and caches installed gems automatically

```

Comment:
Why moving from Jekyll Action to manual build?

Reply:
Because anyways we have to setup build environment to run `jekyll serve`, so running separate action that again sets up ruby and then builds and stores in `out` directory will increase workflow time.